build_all.py

```diff
@@ -0,0 +1,26 @@
+import build_platform as b
+
+# b.execute_build(b.target_macos, b.arch_x86, b.build_all, node_enabled = True, cross_compile = True)
+b.execute_build(b.target_macos, b.arch_x64, b.build_all, node_enabled = True, cross_compile = True)
```

Comment:
The way command-line switches and arguments are translated to changes in the produced binary build artifacts is (roughly) as follows:

`--node-enabled` --> `params.node_enabled` --> `J2V8_NODE_ENABLED` --> `NODE_COMPATIBLE=1`

- `--node-enabled` // passed via the build.py CLI
  - _is parsed in the python code and turned into a boolean variable + value_
- `params.node_enabled` // variable in python containing the parsed boolean value
  - _is put into a "config" dictionary/object that will be passed as a parameter to each build-step function for the build_
  - _each build-step can decide to access the variable and do something with it_
- `J2V8_NODE_ENABLED` // the CMakeLists.txt file exposes this boolean variable that can be overridden when invoking the cmake CLI
  - _the `def build_j2v8_cmake(config):` build-step passes the value of the "config" object property `node_enabled` to [the cmake CLI](https://cmake.org/cmake/help/v2.8.8/cmake.html#opt:-Dvar:typevalue) when it is invoked in this build-step_
- `NODE_COMPATIBLE=1` // if the above CMake variable is set "TRUE"
  - _this compiler preprocessor-definition will be included when generating the native build files_

Having said that, the above original code of `build_all.py` (in the next PR it will be called `build_configs.py`) is just a way to skip over the first step of the chain shown above, and directly pass a user-specified configuration to the python function that then continues to process the build in the same way as if it was originally called from the CLI